pre_commit/languages/docker.py

```diff
@@ -76,7 +78,24 @@ def install_environment(
         os.mkdir(directory)
 
 
+@functools.lru_cache(maxsize=1)
+def docker_is_rootless() -> bool:
+    returncode, stdout, stderr = cmd_output(
+        ('docker', 'system', 'info'),
+    )
+    for line in stdout.splitlines():
+        # rootless docker has "rootless"
+        # rootless podman has "rootless: true"
+        if line.strip().startswith('rootless'):
+            if 'false' not in line:
+                return True
+            break
```

Comment:
this is probably slightly more elegant:
```suggestion
        if not line.strip().startswith('rootless'):
            continue
        return 'false' not in line
```
(though, you don't have to apply it if you prefer the current solution)

Reply:
IMO I found this harder to understand.

Reply:
OK, let's keep the first one then.